### PR TITLE
add elixir 1.9.4

### PR DESCRIPTION
**Pull request by bollwyvl**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by bollwyvl** [comment]
in lieu of @conda-forge/help-erlang , any thoughts @conda-forge/erlang ?

**Comment by bollwyvl** [comment]
> in lieu of @conda-forge/help-erlang , any thoughts @conda-forge/erlang ?

Follow-on: any interest in co-maintaining?

**Comment by bollwyvl** [comment]
Some background: trying to work up to an [IElixir](https://github.com/pprzetacznik/IElixir) binder. I've been able to use a local build of this recipe, but it looks like `IElixir` (and most other elixir things) need `hex` and `rebar` to be used effectively... and then require compilers and proper setting of various `*FLAGS` once you stray off the "pure" elixir path.

```bash
export LDFLAGS="-I${CONDA_PREFIX}/include -L${CONDA_PREFIX}/lib ${LDFLAGS}"
export CPPFLAGS="-I${CONDA_PREFIX}/include -L${CONDA_PREFIX}/lib ${CPPFLAGS}"
export CFLAGS="-I${CONDA_PREFIX}/include -L${CONDA_PREFIX}/lib ${CFLAGS}"
```

I guess if this gets in, those (and potentially their deps) as `elixir-*` would be needed, and something (either `elixir`, `erlang`, `hex` or `rebar`) would need the `FLAGS` cooked in somehow.

**Comment by bollwyvl** [comment]
`mix` also puts stuff in `~/.mix` and `~/.cache/rebar3`. Not sure what to do about those.

File: recipes/elixir/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  version: {{ version }}
+
+source:
+  url: https://github.com/elixir-lang/elixir/archive/v{{ version }}.tar.gz
+  sha256: f3465d8a8e386f3e74831bf9594ee39e6dfde6aa430fe9260844cfe46aa10139
+
+build:
+  number: 0
+  skip: true  # [win]
+  script: make install
```
Comment (scopatz) [inline recipes/elixir/meta.yaml]: Does this need `--prefix=$PREFIX`?

**Comment by bollwyvl** [comment]
ooooh.... that would probably help. Had some issues finding lib/includes
stuff when testing it..

File: recipes/elixir/meta.yaml (hunk 1 above)
Comment (bollwyvl) [reply recipes/elixir/meta.yaml]: Do i have to worry about the other FLAGs i wonder? I'm such a `make` charlatan...

Comment (scopatz) [reply recipes/elixir/meta.yaml]: This is usually enough, unless you are putting stuff in weird places ;)

File: recipes/elixir/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  version: {{ version }}
+
+source:
+  url: https://github.com/elixir-lang/elixir/archive/v{{ version }}.tar.gz
+  sha256: f3465d8a8e386f3e74831bf9594ee39e6dfde6aa430fe9260844cfe46aa10139
+
+build:
+  number: 0
+  skip: true  # [win]
+  script: make prefix={{ PREFIX }} install
```
Comment (scopatz) [inline recipes/elixir/meta.yaml]: This should be `${PREFIX}` so it resolves at build time

Comment (bollwyvl) [reply recipes/elixir/meta.yaml]: Gah, thanks... so used to trying to use the variables to be windows-friendly...

**Comment by scopatz** [comment]
Thanks!

**Comment by bollwyvl** [comment]
Woo... now to figure out how to package `IElixir`, which, surprise, has it's _own_ package manager... thoughts on `hex`, `rebar`?

**Pull request merged**